### ISSUE TITLE
Update animations w/ react-plotly

### ISSUE THREAD

**Issue by BG687** [plotly/react-plotly.js#101]
I would like to animate my stacked area chart when the data updates (so the transition from the old data to the new data is smooth). Right now, when the data updates the chart just instantly changes to reflect the new data. 

I don't see an animate function on the Plotly react component or its prototype. What is the best way to go about animating these charts? Is this possible? 

Thanks

**Comment by nicolaskruchten** [comment]
This isn't currently possible within `react-plotly.js` but we're working on adding support for it in the underlying plotly.js library: https://github.com/plotly/plotly.js/issues/1849

**Comment by GrahamWilsdon** [comment]
@nicolaskruchten any update on this?

**Comment by CarlosAmaral** [comment]
@nicolaskruchten Do we already have Bar chart animations?

**Comment by nicolaskruchten** [comment]
Bar chart transitions are supported in recent versions of Plotly.js, yes.

**Comment by CarlosAmaral** [comment]
@nicholas-esterer I used the same code as @billyhunt. Didn't work. Do you have an example that you can share? Thank you!

**Comment by nicolaskruchten** [comment]
@CarlosAmaral which version of plotly.js are you using? Try 1.54.2 :)

**Comment by CarlosAmaral** [comment]
@nicholas-esterer I did that. Same outcome. No animation.

**Comment by nicolaskruchten** [comment]
Can you provide a full running example in a codepen or something for me to help you debug?

**Comment by qidian99** [comment]
Is there any example of animating the scatter plot?

**Comment by CarlosAmaral** [comment]
@nicolaskruchten https://264b4.codesandbox.io/ sorry for the delay

**Comment by nicolaskruchten** [comment]
@CarlosAmaral thanks! What part of this is meant to be animated?

**Comment by CarlosAmaral** [comment]
@nicolaskruchten good question, the bars going up?

**Comment by nicolaskruchten** [comment]
You mean an on-load transition where the bars grow from zero? Or in response to user input or... ?

**Comment by CarlosAmaral** [comment]
@nicolaskruchten yes, on loading, just making the bars go from zero their value.

**Comment by joan780** [comment]
@nicolaskruchten I run into the same problem. Is there any update?

**Comment by CarlosAmaral** [comment]
It actually works if you switch data. In my case: I was expecting an animation when opening a modal, probably it rendered before.

**Comment by joan780** [comment]
I also need animation when the chart is loading at first. Do you happen to have examples to share?

**Comment by CarlosAmaral** [comment]
@joan780 No, that's something that I wasn't yet able to accomplish but I'm looking into that.

**Comment by joan780** [comment]
@CarlosAmaral I just figured out a way of doing it but probably not a best practice though. Here it is - 
https://codesandbox.io/s/red-shape-bb5y8?file=/src/plotly/plotly.js
Would like feedback about it!

**Comment by tony** [comment]
@joan780 When I try the example, I get this:

`TypeError: Cannot read property 'document' of undefined`

![image](https://user-images.githubusercontent.com/26336/103463190-434e5080-4cf0-11eb-86ba-ead834085cde.png)

Not sure if it's related to the code itself, or codesandbox

**Comment by AlexBlack2202** [comment]
any update on this?

**Comment by openSourcerer9000** [comment]
> Is there any example of animating the scatter plot?

Following this - it would be great to have a starting place to go off

**Comment by Sid110307** [comment]
any updates?
